bundles/core/org.eclipse.smarthome.core/src/main/java/org/eclipse/smarthome/core/scheduler/AbstractExpression.java

```diff
@@ -207,22 +207,51 @@ protected void prune() {
 
     @Override
     public Date getTimeAfter(Date afterTime) {
+
+        Date currentStartDate = getStartDate();
+
+        if (hasFloatingStartDate()) {
+            try {
+                clearCandidates();
+                setStartDate(afterTime);
+            } catch (IllegalArgumentException | ParseException e) {
+                logger.error("An exception occurred while setting the start date : '{}'", e.getMessage());
+            }
+        }
+
         if (getCandidates().isEmpty()) {
```

Comment:
We can use a `else if` here. If it is a foating start date, set start date has been already called. If this results in an empty candidates list, is there any reason to call it again?